cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/EOGBreaker.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph
+
+/**
```

Comment:
I wonder if it would be better to iterate the AST, but order them (partially) in the EOG, maybe using the lattice? So that we really catch all nodes?

Reply:
I'm not completely sure what you mean with it but what we could do in the `SubgraphWalker` is to add AST-children if they are disconnected from the EOG. But this sounds to me like a solution to address

1) the case where we cannot precisely model the EOG (e.g. because it's evaluated at a later point in time or it depends on an uncertain trigger),
2) the case where there should be an EOG but we forgot to model it in the EOG Pass and
3) there case where a node is not in EOG but for some reason it holds relevant information but we never explicitly thought of it being relevant (What is this?! Should never happen because it would somewhat mean that we don't know what we're doing) ;)

Numbers 2 and 3 simply shouldn't happen and I'd prefer to see a test failing if that's the case

The lattice-based fixed-point iteration won't help us in this case but in fact, I should check that we do not miss visiting some parts there

Reply:
I added the resolution start holder interface also to the TryStatement, so that resolution starts with all catch clauses. That seems to work for now. This way we don't need an extra interface